driver-core/src/main/com/mongodb/internal/connection/ByteBufBsonDocument.java

```diff
@@ -51,9 +52,11 @@
             documentsBuffer.position(documentsBuffer.position() - 4);
             ByteBuf documentBuffer = documentsBuffer.duplicate();
             documentBuffer.limit(documentBuffer.position() + documentSizeInBytes);
-            documents.add(new ByteBufBsonDocument(documentBuffer));
+            documents.add(new ByteBufBsonDocument(new ByteBufNIO(documentBuffer.asNIO())));
```

Comment:
Interesting - this is different to what I see.

I have to release the buffer after - otherwise this causes a leak as the ref count for the `documentBuffer` is 1.  So this allows access to the underlying data (while it exists).

If I just do `documents.add(new ByteBufBsonDocument(documentBuffer));` and then release the `documentBuffer` any access to the buffer causes a `llegalReferenceCountException: refCnt: 0, increment: 1` exception

Reply:
` documents.add(new ByteBufBsonDocument(new ByteBufNIO(documentBuffer.asNIO())));` is needed so I can access the document both using netty and nio2.  

If I don't release `documentBuffer` then netty will leak. If I do release `documentBuffer` and use `new ByteBufBsonDocument(documentBuffer)` nio2 can NPE in the command listener.

Reply:
OK, I see it now.  It happens in the commandSucceeded when accessing the response, but not in commandStarted when accessing the command